### PR TITLE
Docker Userspace Remapping Options

### PR DESCRIPTION
**Pull request by gambol99**
- adding the userspace remap options for docker daemon

### CODE REVIEW

**Comment by KashifSaadat** [comment]
Holding fire on this just yet, needs some testing :)

File: pkg/apis/kops/dockerconfig.go [hunk 1]
```diff
@@ -50,6 +50,8 @@ type DockerConfig struct {
 	Storage *string `json:"storage,omitempty" flag:"storage-driver"`
 	// StorageOpts is a series of options passed to the storage driver
 	StorageOpts []string `json:"storageOpts,omitempty" flag:"storage-opt,repeat"`
+	// UserspaceRemap sets the userspace remapping option for the docker daemon
+	UserspaceRemap string `json:"userspaceRemap,omitempty" flag:"userns-remap"`
```
Comment (justinsb) [inline pkg/apis/kops/dockerconfig.go]: Can we name the field `UserNamespaceRemap` and json `userNamespaceRemap`?  Userspace means something else to me :-)

**Comment by justinsb** [comment]
I suggested a different name (the json is what matters).

Because this is just mapping a flag and is optional, no reason not to merge this though once we agree on a name!

File: pkg/apis/kops/dockerconfig.go (hunk 1 above)
Comment (gambol99) [reply pkg/apis/kops/dockerconfig.go]: yep ... make's sense! :-)

**Comment by gambol99** [comment]
@justinsb .. changed the name and json tags to UserNamespaceRenap / userNamespaceRemap and rebased as suggested ..

**Comment by justinsb** [comment]
OK - will target to 1.12

If you'd rather not rebase, this has no impact so we can really merge whenever we want - feel free to self lgtm

**Pull request merged**